xarray/core/dataarray.py

```diff
@@ -256,23 +258,6 @@ def _replace_maybe_drop_dims(self, variable, name=__default):
                                  if set(v.dims) <= allowed_dims)
         return self._replace(variable, coords, name)
 
-    def _replace_indexes(self, indexes):
```

Comment:
If I remember well, we used duplicates to avoid using `_to_temp_dataset` and `_from_temp_dataset` in `__getitem__`. But now that `_replace_indexes` has more logic implemented, maybe it is a good idea to reduce duplicates?